src/tensors/adjoint.jl

```diff
@@ -6,100 +6,38 @@
 Specific subtype of [`AbstractTensorMap`](@ref) that is a lazy wrapper for representing the
 adjoint of an instance of [`AbstractTensorMap`](@ref).
 """
-struct AdjointTensorMap{T,S,N₁,N₂,TT<:AbstractTensorMap{T,S,N₂,N₁}} <:
+struct AdjointTensorMap{T,S,N₁,N₂,TT<:AbstractTensorMap{T,S,N₁,N₂}} <:
        AbstractTensorMap{T,S,N₁,N₂}
     parent::TT
 end
 
-#! format: off
-const AdjointTrivialTensorMap{T,S,N₁,N₂,TT<:TrivialTensorMap{T,S,N₂,N₁}} =
-    AdjointTensorMap{T,S,N₁,N₂,TT}
-#! format: on
-
 # Constructor: construct from taking adjoint of a tensor
+Base.adjoint(t::AdjointTensorMap) = t.parent
 Base.adjoint(t::AbstractTensorMap) = AdjointTensorMap(t)
-Base.adjoint(t::AdjointTensorMap) = parent(t)
-
-Base.parent(t::AdjointTensorMap) = t.parent
```

Comment:
I usually find it a little clearer to read code in terms of `t` and `parent(t)` instead of `t` and `t'`, as the latter is a bit confusing since `adjoint(t)` is actually the non-adjoint version. No hard opinions here though

Reply:
I saw some Github issue or discourse post of someone complaining that the semantics of parent were not well defined or something (I did not read the thread). That's why I tried to avoid it, but I will revert this.